src/runtime/bootstrap-lazy.ts

```diff
@@ -185,16 +185,18 @@ export const bootstrapLazy = (lazyBundles: d.LazyBundlesRuntimeData, options: d.
     });
   });
 
+  dataStyles.innerHTML = SLOT_FB_CSS;
+  dataStyles.setAttribute('data-styles', '');
+  head.insertBefore(dataStyles, metaCharset ? metaCharset.nextSibling : head.firstChild);
+
   if (BUILD.invisiblePrehydration && (BUILD.hydratedClass || BUILD.hydratedAttribute)) {
-    visibilityStyle.innerHTML = cmpTags + HYDRATED_CSS;
-    visibilityStyle.setAttribute('data-styles', '');
+    dataStyles.innerHTML += cmpTags + HYDRATED_CSS;
+  }
 
-    // Apply CSP nonce to the style tag if it exists
-    const nonce = plt.$nonce$ ?? queryNonceMetaTagContent(doc);
-    if (nonce != null) {
-      visibilityStyle.setAttribute('nonce', nonce);
-    }
-    head.insertBefore(visibilityStyle, metaCharset ? metaCharset.nextSibling : head.firstChild);
+  // Apply CSP nonce to the style tag if it exists
+  const nonce = plt.$nonce$ ?? queryNonceMetaTagContent(doc);
+  if (nonce != null) {
+    dataStyles.setAttribute('nonce', nonce);
```

Comment:
This shouldn't have been contingent on the `invisiblePrehydration` stuff in the first place